### PR TITLE
Bugfix: CollectIlluminaLaneMetricsTest when converting the read struc…

### PR DESCRIPTION
**Pull request by nh13**
…ture.

It was not converting the read structure properly; now with tests.

I don't know why I didn't have unit tests for this previously.  Weird.

### CODE REVIEW

File: tasks/src/main/scala/dagr/tasks/picard/CollectIlluminaLaneMetrics.scala [hunk 1]
```diff
@@ -29,6 +29,23 @@ import dagr.tasks.DagrDef.DirPath
 
 import scala.collection.mutable.ListBuffer
 
+object CollectIlluminaLaneMetrics {
+  def convertBarcodeAndSkipToTemplate(readStructure: String): String = {
+    readStructure
```
Comment (tfenne) [inline tasks/src/main/scala/dagr/tasks/picard/CollectIlluminaLaneMetrics.scala:34]: I'm finding this super hard to follow - but there are tests so if you're happy I'm happy.

I think if I were to want to understand this in the future I would break it up a bit and give the intermediates really clear names and types so I could see what's going on.

File: tasks/src/test/scala/dagr/tasks/picard/CollectIlluminaLaneMetricsTest.scala [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * THE SOFTWARE.
+ *
+ */
+
+package dagr.tasks.picard
+
+import dagr.commons.util.UnitSpec
+
+class CollectIlluminaLaneMetricsTest extends UnitSpec {
+  "CollectIlluminaLaneMetrics.convertBarcodeAndSkipToTemplate" should "convert barcodes and skips to template" in {
```
Comment (tfenne) [inline tasks/src/test/scala/dagr/tasks/picard/CollectIlluminaLaneMetricsTest.scala:31]: Yay tests!

**Review by tfenne — APPROVED** [review]

File: tasks/src/main/scala/dagr/tasks/picard/CollectIlluminaLaneMetrics.scala (hunk 1 above)
Comment (nh13) [reply tasks/src/main/scala/dagr/tasks/picard/CollectIlluminaLaneMetrics.scala:34]: I find the `foldLeft` the most incomprehensible, and didn't find a way I liked to split it up.  It is what it is.
